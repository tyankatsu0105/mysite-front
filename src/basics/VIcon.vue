<template>
	<component
		:is="icon"
		class="VIcon"
	/>
</template>

<script>
import IconHatena from "@/basics/VIcon/IconHatena.vue";
export default {
  components: {
    "icon-hatena": IconHatena
  },
  props: {
    name: {
      type: String,
      requirer: true,
      default: ""
    }
  },
  computed: {
    icon() {
      return "icon-" + this.name.toLowerCase();
    }
  }
};
</script>

<style scoped lang='scss'>
.VIcon {
  svg {
    color: currentColor;
  }
  /deep/ g {
    fill: currentColor;
  }
}
</style>
