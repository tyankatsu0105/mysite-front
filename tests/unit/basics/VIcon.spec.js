import { shallowMount } from "@vue/test-utils";
import VIcon from "@/basics/VIcon.vue";

describe("VIcon", () => {
  it("render", () => {
    const wrapper = shallowMount(VIcon, {
      slots: {
        default: '<svg class="icon"></svg>'
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
