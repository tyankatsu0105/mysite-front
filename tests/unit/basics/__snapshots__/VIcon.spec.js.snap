// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VIcon render 1`] = `<div class="VIcon"><svg class="icon"></svg></div>`;
